### PR TITLE
PLANET-5757 jQuery: remove jQuery from global.js

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
Removed global.js file and added title attribute in twig file & in PHP file by altering original output

Ref: [JIRA 5757](https://jira.greenpeace.org/browse/PLANET-5757)

---

**Testing steps:**
- Add native image block on page/post/campaign and test the frontend page the image has a title attribute
- The same fix is added for some other pages like author profile page, search, taxonomy page etc.

### CODE REVIEW

File: src/MasterSite.php [hunk 1]
```diff
@@ -1248,16 +1248,27 @@ public function add_image_attachment_fields_to_save( $post, $attachment ) {
...
+		$img_post_meta = $image_id ? get_post_meta( $image_id ) : [];
+		if ( ! $img_post_meta ) {
+			return $content;
+		}
+
+		$credit   = $img_post_meta[ self::CREDIT_META_FIELD ][0] ?? '';
+		$alt_text = $img_post_meta['_wp_attachment_image_alt'][0] ?? '';
+
+		if ( $alt_text ) {
+			$content = str_replace( 'alt=', 'title="' . esc_attr( $alt_text ) . '" alt=', $content );
```
Comment (Inwerpsel) [inline src/MasterSite.php:1269]: I think we can match `' alt='` too (so with preceding space), which would decrease the chance of matching unintended things.

File: templates/tease-author.twig [hunk 2]
```diff
@@ -3,6 +3,7 @@
 		class="d-flex search-result-item-image"
 		src="{{ post.thumbnail.src('thumbnail') }}"
 		alt="{{ fn('esc_attr', post.thumbnail.alt|default( post.title ) )|raw }}"
+		title="{{ fn('esc_attr', post.thumbnail.alt|default( post.title ) )|raw }}"
```
Comment (Inwerpsel) [inline templates/tease-author.twig:6]: This logic is a bit complex and with this change would occur 6 times in 3 files, maybe it's better to write a function for it, or alternative maybe assign it as a template variable? The only difference is [this other one uses `post.post_title`](https://github.com/greenpeace/planet4-master-theme/pull/1395/files#diff-f721ebbf1a207a8e1119c863660a9ec1cac8e82c0e5dcbccebbef1721b41807bR26), but maybe all of them could use `post.title`.

**Review by Inwerpsel — APPROVED** [review]
Looks good, left some suggestions. There could be some rare cases where the previous code was adding the title that are not covered by these changes. For example images in a custom HTML block. But can't think of anything that's frequent enough to be blocking. :+1:

File: src/MasterSite.php [hunk 3]
```diff
@@ -1248,16 +1248,27 @@ public function add_image_attachment_fields_to_save( $post, $attachment ) {
...
-	 * @return string HTML content of image element with credit field in caption.
+	 * @return string HTML content of image element with credit field in caption and alt text in image title.
 	 */
 	public function p4_core_image_block_render( $attributes, $content ) {
-		$image_id = isset( $attributes['id'] ) ? trim( str_replace( 'attachment_', '', $attributes['id'] ) ) : '';
-		$credit   = $image_id ? get_post_meta( $image_id, self::CREDIT_META_FIELD, true ) : '';
+		$image_id      = isset( $attributes['id'] ) ? trim( str_replace( 'attachment_', '', $attributes['id'] ) ) : '';
+		$img_post_meta = $image_id ? get_post_meta( $image_id ) : [];
+		if ( ! $img_post_meta ) {
+			return $content;
```
Comment (Inwerpsel) [inline src/MasterSite.php:1262]: Sonarcloud complains about this 4th return statement that's added (it only allows 3). Though in this case the added return does make sense, so not sure how we should please Sonar :stuck_out_tongue:

File: src/MasterSite.php [hunk 4]
```diff
@@ -1248,16 +1248,27 @@ public function add_image_attachment_fields_to_save( $post, $attachment ) {
...
+		}
+
+		$credit   = $img_post_meta[ self::CREDIT_META_FIELD ][0] ?? '';
+		$alt_text = $img_post_meta['_wp_attachment_image_alt'][0] ?? '';
+
+		if ( $alt_text ) {
+			$content = str_replace( 'alt=', 'title="' . esc_attr( $alt_text ) . '" alt=', $content );
+		}
+
 		if ( empty( $image_id ) || empty( $credit ) ) {
```
Comment (Inwerpsel) [inline src/MasterSite.php:1272]: With the new check you added earlier, it's now impossible that it gets here and `empty( $image_id )` would be true, so I guess we can remove that part of the condition.

File: templates/tease-author.twig (hunk 2 above)
Comment (sagarsdeshmukh) [reply templates/tease-author.twig]: For now use a template variable :)

File: src/MasterSite.php (hunk 3 above)
Comment (sagarsdeshmukh) [reply src/MasterSite.php:1262]: Removed the one return statement, hope Sonarcloud is happy now 😄

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [venus](https://www-dev.greenpeace.org/test-venus)

:watch: 2021.06.07 12:46:22

**Comment by sagarsdeshmukh** [comment]
Hi @Inwerpsel , I updated the PR after last review, could you please take a look on new changes. 🙂

File: templates/blocks/site_logo.twig [hunk 5]
```diff
@@ -5,6 +5,7 @@
 		<img
 			src="{{ data_nav_bar.images }}gp-logo-{{ logo_color }}.svg"
 			alt="Greenpeace"
+			title="Greenpeace"
```
Comment (comzeradd) [inline templates/blocks/site_logo.twig:8]: I don't think we need a tooltip in the site logo, the author profile image, or even the article thumbnails on search and articles block (the other PR). 

According to the [original ticket](https://jira.greenpeace.org/browse/PLANET-1757) the intention was mostly to add tooltips on images being included on posts and pages so users can easily see credit on hover. I don't think we should expand the scope of the initial code.

Comment (sagarsdeshmukh) [reply templates/blocks/site_logo.twig:8]: Yes got the point 👍, It was misunderstood by me as applicable over all images.

I will update both the PR accordingly 🙂

Comment (sagarsdeshmukh) [reply templates/blocks/site_logo.twig]: Both PRs updated, please have a look.

**Review by comzeradd — APPROVED** [review]
:+1:

**Pull request merged**